src/keyboard-x11.c

```diff
@@ -380,19 +380,29 @@ void keyboard_SetLayout(Keyboard *pKeyboard, gint nLayout)
     }
     else
     {
-        // TODO
+        gchar *sDisplay = ":0";
+        Display *pDisplay = XOpenDisplay (sDisplay);
+
+        if (pDisplay == NULL)
+        {
+            sDisplay = ":1";
```

Comment:
I will look into this, however, I think we might eventually need to offload certain tasks to Arctica, since some applications/commands simply cannot be run properly from within indicators.

Having done the framework for the a11y indicator, it seems to me that the most elegant way would be ask Actica over D-Bus to change the layout for us, start Orca/Onboard, change the theme, etc.

I'll revisit my suggestion after I see what `dbus-update-activation-environment` can do for us.

Reply:
Fixed.